ai-assistant/src/ai/prompts.ts

```diff
@@ -22,6 +22,55 @@ RESOURCE CREATION GUIDELINES:
+- ALWAYS mention what specific change you're making (e.g., "Setting replica count to 3")
+- Then provide ONLY the patch object that needs to be merged, NOT the full resource YAML
+- Use kubernetes_api_request with method="PUT" and provide only the patch in the body
+- The system will automatically merge your patch with the current resource and make a PUT request with the complete updated resource
+- Example: "change replica count to 3" → Show "Setting replicas to 3", then call kubernetes_api_request with PUT method and body: {"spec": {"replicas": 3}}
+- Example: "remove liveness probe from deployment" → Show "Removing liveness probe from container 'headlamp'", then call kubernetes_api_request with PUT method and body: {"spec": {"template": {"spec": {"containers": [{"name": "headlamp", "livenessProbe": null}]}}}}
+- Example: "remove security context" → Show "Removing security context from container", then call kubernetes_api_request with PUT method and body: {"spec": {"template": {"spec": {"containers": [{"name": "headlamp", "securityContext": null}]}}}}
+- IMPORTANT: For Pods with immutable fields, suggest updating the controlling resource instead
+- For managed Pods (created by controllers), always update the controller, not the Pod directly
+- MANAGED POD DETECTION: Check Pod's ownerReferences to identify the controller:
+  - Deployment-managed Pods: typically have names like "name-hash-hash" (e.g., "headlamp-764477b977-5sv8s")
+  - ReplicaSet-managed Pods: similar naming pattern "name-hash"
+  - StatefulSet-managed Pods: have ordinal names like "name-0", "name-1", etc.
+  - DaemonSet-managed Pods: typically include node name in suffix
+  - Job-managed Pods: have names like "job-name-hash"
+  - CronJob-managed Pods: have names like "cronjob-name-timestamp-hash"
+- When updating managed Pods, identify and update the appropriate controller (Deployment, StatefulSet, DaemonSet, ReplicaSet, Job, CronJob)
+
+UPDATE EXAMPLES:
+To remove liveness probe from a deployment:
+1. Show: "Removing liveness probe from container 'headlamp'"
+2. Use PUT with patch body: {"spec": {"template": {"spec": {"containers": [{"name": "headlamp", "livenessProbe": null}]}}}}
+
+To change replica count:
+1. Show: "Setting replica count to 3"
+2. Use PUT with patch body: {"spec": {"replicas": 3}}
+
+To remove security context:
+1. Show: "Removing security context from container 'headlamp'"
+2. Use PUT with patch body: {"spec": {"template": {"spec": {"containers": [{"name": "headlamp", "securityContext": null}]}}}}
+
+To update an environment variable:
+1. Show: "Updating environment variable ENV_VAR to 'new-value'"
+2. Use PUT with patch body: {"spec": {"template": {"spec": {"containers": [{"name": "headlamp", "env": [{"name": "ENV_VAR", "value": "new-value"}]}]}}}}
+
+To add a label:
+1. Show: "Adding label 'environment: production'"
+2. Use PUT with patch body: {"metadata": {"labels": {"environment": "production"}}}
```

Comment:
Maybe this belongs to the Kubernetes tool? We cannot have everything in the base prompt. The tools should provide the solutions for specific things like these.

Reply:
moved this k8s tool